docs/application/native/guides/text-input/voice-control-manager.md

```diff
@@ -0,0 +1,1174 @@
+  This function is not used when the voice control service is in the `VC_SERVICE_STATE_NONE` state:
+
+  ```c
+  void
+  get_current_language()
+  {
+      int ret;
+      char* current_lang = NULL;
+      ret = vc_mgr_get_current_language(&current_lang);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get the microphone volume during recording using the `vc_mgr_get_recroding_volume()` function.
+  The recording volume value is retrieved periodically with the short-term recorded sound data as decibels (dB).
+  The recording volume normally has a negative value, and 0 is the maximum value.
+  This function is used when the voice control service is in the `VC_SERVICE_STATE_RECORDING` state:
+
+  ```c
+  void get_recording_volume()
+  {
+      int ret;
+      float recording_volume = 0.0f;
+      ret = vc_mgr_get_recording_volume(&recording_volume);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get or set the audio in type using the `vc_mgr_get_audio_type()` and `vc_mgr_set_audio_type()` functions.
+  The audio type values can be `VC_AUDIO_ID_BLUETOOTH` or `VC_AUDIO_ID_MSF` in string.
+  These functions are used when the voice control manager is in the `VC_STATE_READY` state:
+
+  ```c
+  /* Get */
+  void get_audio_type()
+  {
+      int ret;
+      char* audio_id = NULL;
+      ret = vc_mgr_get_audio_type(&audio_id);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Set */
+  void set_audio_type(const char* audio_id)
+  {
+      int ret;
+      ret = vc_mgr_set_audio_type(audio_id);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get or set the recognition mode using the `vc_mgr_get_recognition_mode()` and `vc_mgr_set_recognition_mode()` functions.
+  The default value of recognition mode is `VC_RECOGNITION_MODE_STOP_BY_SILENCE`. If you want to set the manual mode, you can use `VC_RECOGNITION_MODE_MANUAL`.
+  These functions are used when the voice control manager is in the `VC_STATE_READY` state:
+
+  ```c
+  /* Get */
+  void get_recognition_mode()
+  {
+      int ret;
+      vc_recognition_mode_e mode;
+      ret = vc_mgr_get_recognition_mode(&mode);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+
+  /* Set */
+  void set_recognition_mode(vc_recognition_mode_e mode)
+  {
+      int ret;
+      ret = vc_mgr_set_recognition_mode(mode);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get or set private data between voice control manager and voice control engine using the `vc_mgr_get_private_data()` and `vc_mgr_set_private_data()` functions.
+  `vc_mgr_get_private_data()` is used when the parameters move from voice control engine to voice control manager.
+  `vc_mgr_set_private_data()` is used when the parameters move from voice control manager to voice control engine.
+  These functions are used when the voice control manager is in the `VC_STATE_READY` state:
```

Comment:
Parameters move ? can you please let me know what is the intention here? 

I just gave a suggestion, kindly verify. 


```suggestion 
- Get or set private data between voice control manager and voice control engine using `vc_mgr_get_private_data()` and `vc_mgr_set_private_data()`.
  The `vc_mgr_get_private_data()` is used when the parameters are passed from the voice control engine to the voice control manager and vice versa. These functions are used when the voice control manager is in the `VC_STATE_READY` state:
```

Reply:
Using `vc_mgr_get_private_data()`, the voice control manager gets the `data` of `private_key`. It means that the voice control engine sends the `data` of `private_key`.
And the voice control manager sets the `data` of `private_key` using `vc_mgr_set_private_data()`. By this function, the voice control manager sends the `data` to the voice control engine.